api/src/main/java/org/eclipse/microprofile/reactive/messaging/connector/IncomingConnectorFactory.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2018-2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.reactive.messaging.connector;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.MessagingProvider;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+
+import java.util.NoSuchElementException;
+
+/**
+ * SPI used to implement a connector managing a source of messages for a specific <em>transport</em>. Typically, to
+ * handle the consumption of records from Kafka, the reactive messaging extension would need to implement a {@code bean}
+ * implementing this interface. This bean is called for every {@code stream} that need to be created for this specific
+ * <em>transport</em> (so Kafka in the example). These streams are connected to methods annotated with
+ * {@link org.eclipse.microprofile.reactive.messaging.Incoming}.
+ * <p>
+ * The factory is called to create a {@code stream} for each configured <em>transport</em>. The configuration is done using
+ * MicroProfile Config. The following snippet gives an example for a hypothetical Kafka connector:
+ *
+ * <pre>
+ * mp.messaging.incoming.my-stream.type=i.e.m.reactive.messaging.impl.kafka.Kafka
+ * mp.messaging.incoming.my-stream.bootstrap-servers=localhost:9092
+ * mp.messaging.incoming.my-stream.topic=my-topic
+ * ...
+ * </pre>
+ * <p>
+ * This configuration keys are structured as follows: {@code mp.messaging.[incoming|outgoing].stream-name.attribute}.
```

Comment:
"This" -> "The"

Reply:
"The portion of the key that precedes the {@code attribute} is acts as a property prefix
that has a common structure across all {@code streams}."
(to introduce the term 'prefix' that is used below)

Reply:
...of course we are only using 'incoming' here.
Is this 'strictly' true, requiring the duplicate of some keys perhaps?